app/routes/me.tsx

```diff
@@ -61,64 +69,133 @@ export const action: ActionFunction = async ({request}) => {
-        </Form>
-      </div>
-      <details>
-        <summary>Change account details</summary>
-
-        <Form method="post" action="/me">
-          <input type="hidden" name="actionId" value="change details" />
-          <div>
-            <label htmlFor="firstName">First Name</label>
-            <input
-              id="firstName"
-              name="firstName"
-              defaultValue={user.firstName}
-            />
+    <Form action="/me" method="post" className="mt-12">
+      <Grid>
+        <div className="col-span-full mb-12 lg:mb-20">
+          <div className="flex items-center justify-between">
+            <div>
+              <H2 className="mb-2">Here’s your profile.</H2>
+              <H2 variant="secondary" as="p">
+                Edit as you wish.
+              </H2>
+            </div>
+            <Link
+              to="/logout"
+              className="flex col-span-full items-center px-11 py-6 text-black dark:text-white border-2 border-gray-600 rounded-full space-x-4 lg:col-span-8 lg:col-start-3"
+            >
+              <LogoutIcon />
+              <H6 as="span">logout</H6>
+            </Link>
           </div>
-          <button type="submit">Submit</button>
-        </Form>
-      </details>
-      <div>
-        {user.discordId ? (
-          <div>
-            Connected to discord account:{' '}
-            <a href={`https://discord.com/users/${user.discordId}`}>
-              {userInfo.discord?.username ?? user.discordId}
-            </a>
+        </div>
+
+        <div className="col-span-full mb-24 lg:col-span-5 lg:mb-0">
+          <Label className="mb-4" htmlFor="firstName">
+            First name
+          </Label>
+
+          <Input
+            className="mb-8"
+            name="firstName"
+            id="firstName"
+            autoComplete="firstName"
+            defaultValue={user.firstName}
+            required
+          />
+
+          <Label className="mb-4" htmlFor="email-address">
+            Email address
+          </Label>
+
+          <Input
+            name="email"
+            id="email-address"
+            autoComplete="email"
+            required
+            defaultValue={user.email}
+            className="mb-8"
+          />
+
+          <div className="flex flex-wrap items-baseline justify-between mb-4">
+            <Label htmlFor="discord-id">Discord</Label>
+            <p
+              id="discord-message"
+              className="dark:text-blueGray-500 text-gray-500 text-lg"
+            >
+              {user.discordId ? (
+                <a
+                  className="text-black dark:text-white"
+                  href={`https://discord.com/users/${user.discordId}`}
+                >
+                  connected
+                </a>
+              ) : (
+                <a className="text-black dark:text-white" href={authorizeURL}>
+                  Connect to Discord
+                </a>
+              )}
+            </p>
```

Comment:
This was my own creativity, as the Discord input wasn't included in the design. I think it looks nice with the connect link there on the top right. What do you think?

Reply:
Yup, I like this. Eventually I think I'll get the user's discord avatar and put it in there somehow. I'll also want a way to disconnect. But I like what you've got here for now 👍

Reply:
When you add more options, we could add those under a separate section. Just remove this Discord input, and add new headings with all Discord stuff below that. A bit like how `/signup` separates team selection from first name.